Firestore/Example/Tests/Local/FSTLevelDBBenchmarkTests.mm

```diff
@@ -0,0 +1,145 @@
+    FillDB();
+  }
+
+  virtual void TearDown(benchmark::State &state) {
+    _db = nil;
+  }
+
+  void FillDB() {
+    LevelDbTransaction txn(_db.ptr.get(), "benchmark");
+
+    for (int i = 0; i < _numDocuments; i++) {
+      FSTDocumentKey *docKey =
+          [FSTDocumentKey keyWithPathString:[NSString stringWithFormat:@"docs/doc_%i", i]];
+      std::string docKeyString = [FSTLevelDBRemoteDocumentKey keyWithDocumentKey:docKey];
+      txn.Put(docKeyString, DocumentData());
+      WriteIndex(txn, docKey);
+    }
+    txn.Commit();
+    // Force a write to disk to simulate startup situation
+    _db.ptr->CompactRange(NULL, NULL);
+  }
+
+ protected:
+  void WriteIndex(LevelDbTransaction &txn, FSTDocumentKey *docKey) {
+    txn.Put([FSTLevelDBDocumentTargetKey keyWithDocumentKey:docKey targetID:_targetID],
+            _emptyBuffer);
+    txn.Put([FSTLevelDBTargetDocumentKey keyWithTargetID:_targetID documentKey:docKey],
+            _emptyBuffer);
+  }
+
+  FSTLevelDB *_db;
+  // Arbitrary target ID
+  FSTTargetID _targetID = 1;
+  int _numDocuments = 10;
+  std::string _emptyBuffer;
+};
+
+// Plan: write a bunch of key/value pairs w/ empty strings (index entries)
+// Write a couple large values (documents)
+// In each test, either overwrite index entries and documents, or just documents
+
+BENCHMARK_DEFINE_F(LevelDBFixture, RemoteEvent)(benchmark::State &state) {
+  bool writeIndexes = (bool)state.range(0);
+  int64_t documentSize = state.range(1);
+  int64_t docsToUpdate = state.range(2);
+  std::string documentUpdate = UpdatedDocumentData((int)documentSize);
+  for (auto _ : state) {
+    LevelDbTransaction txn(_db.ptr.get(), "benchmark");
+    for (int i = 0; i < docsToUpdate; i++) {
+      FSTDocumentKey *docKey =
+          [FSTDocumentKey keyWithPathString:[NSString stringWithFormat:@"docs/doc_%i", i]];
+      if (writeIndexes) WriteIndex(txn, docKey);
+      std::string docKeyString = [FSTLevelDBRemoteDocumentKey keyWithDocumentKey:docKey];
+      txn.Put(docKeyString, documentUpdate);
+    }
+    txn.Commit();
+  }
+}
+
+/**
+ * Adjust ranges to control what test cases run. Outermost loop controls whether or
+ * not indexes are written, the inner loops control size of document writes and number
+ * of document writes.
+ */
+static void TestCases(benchmark::internal::Benchmark *b) {
+  for (int writeIndexes = 0; writeIndexes <= 1; writeIndexes++) {
+    for (int documentSize = 1 << 10; documentSize <= 1 << 20; documentSize *= 4) {
+      for (int docsToUpdate = 1; docsToUpdate <= 5; docsToUpdate++) {
+        b->Args({writeIndexes, documentSize, docsToUpdate});
+      }
+    }
+  }
+}
+
+BENCHMARK_REGISTER_F(LevelDBFixture, RemoteEvent)
+    ->Apply(TestCases)
+    ->Unit(benchmark::kMicrosecond)
+    ->Repetitions(5);
+
+@interface FSTLevelDBBenchmarkTests : XCTestCase
+@end
+
+@implementation FSTLevelDBBenchmarkTests
+
+- (void)testRunBenchmarks {
+  // Enable to run benchmarks.
+  if (false) {
```

Comment:
This is probably better captured as an early return so that this thing ends up written at a normal indent.

Also a better way to make this work is to create a new `Benchmarks_iOS` target within the project and add this test to that target instead of `Firestore_Tests`. That way we don't have to edit the files to run the benchmarks and they naturally exclude themselves during normal work.